src/Compatibility/Microsoft.DotNet.PackageValidation/GetLastStablePackage.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Microsoft.Build.Framework;
+using Microsoft.NET.Build.Tasks;
+using NuGet.Common;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.PackageValidation
+{
+    public class GetLastStablePackage : TaskBase
+    {
+        [Required]
+        public string PackageId { get; set; }
+
+        public string[] NugetFeeds { get; set; }
+
+        public string LocalPackagesPath { get; set; }
+
+        public bool UseLocalPackagesPath { get; set; }
+
+        [Output]
+        public string LastStableVersion { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            NuGetVersion version = null;
+            if (UseLocalPackagesPath)
+            {
+                IEnumerable<NuGetVersion> versions = null;
+                versions = GetLatestPackageFromLocalFeed(LocalPackagesPath);
+                NuGetVersion packageVersion = versions?.Where(t => !t.IsPrerelease).OrderByDescending(t => t.Version).FirstOrDefault();
+                LastStableVersion = packageVersion?.ToNormalizedString();
+                return;
```

Comment:
You want to return a boolean here: https://github.com/dotnet/runtime/blob/268ad5bb91eb163c4cfa1de63d2c939ec397f116/src/tasks/installer.tasks/GenerateFileVersionProps.cs#L160.